src/app/frontend/deploy/deploy.controller.js

```diff
@@ -59,6 +68,8 @@ export default class DeployController {
 
     /** @private {boolean} */
     this.isDeployInProgress_ = false;
+
+    this.initializeNamespaces_($log, $resource);
```

Comment:
I'd refrain from displaying entire form until the namespaces are loaded. What do you think? With that there'll be no flickering in the UI.